### PR TITLE
Add some syscalls to support Nginx

### PR DESCRIPTION
**Pull request by cqs21**
- fcntl, F_SETOWN
- prctl, PR_SET_DUMPABLE
- ioctl, FIONBIO/FIOASYNC

see issue: #960

### CODE REVIEW

**Comment by tatetian** [comment]
@liqinggd Could you review this PR? Thanks!

File: kernel/aster-nix/src/net/socket/ip/datagram/mod.rs [hunk 1]
```diff
@@ -231,6 +237,32 @@ impl FileLike for DatagramSocket {
...
+                let mut flags = self.status_flags();
+                flags.set(StatusFlags::O_NONBLOCK, is_nonblocking);
+                self.set_status_flags(flags)?;
+                Ok(0)
+            }
+            IoctlCmd::FIOASYNC => {
+                let is_async = read_val_from_user::<i32>(arg)? != 0;
+                let mut flags = self.status_flags();
+                // Did FASYNC state change ?
+                if flags.contains(StatusFlags::O_ASYNC).bitxor(is_async) {
```
Comment (liqinggd) [inline kernel/aster-nix/src/net/socket/ip/datagram/mod.rs:253]: The `O_ASYNC` flag does not appear to be fully supported. For example, in status_flags, only `O_NONBLOCK` is supported. So, in reality, this is just a dummy implementation. I suggest adding a TODO to support `O_ASYNC`.

File: kernel/aster-nix/src/net/socket/unix/stream/socket.rs [hunk 2]
```diff
@@ -121,16 +132,60 @@ impl FileLike for UnixStreamSocket {
...
+                }
+                Ok(0)
+            }
+            _ => todo!(),
+        }
+    }
+
+    fn owner(&self) -> Result<Uid> {
+        let pid = *self.owner.read();
+        Ok(Uid::new(pid))
```
Comment (liqinggd) [inline kernel/aster-nix/src/net/socket/unix/stream/socket.rs:171]: Why use `Uid` here?  I think it should be a `Pid`.

File: kernel/aster-nix/src/net/socket/unix/stream/socket.rs [hunk 3]
```diff
@@ -121,16 +132,60 @@ impl FileLike for UnixStreamSocket {
...
+    }
+
+    fn set_owner(&self, pid: Uid) -> Result<()> {
```
Comment (liqinggd) [inline kernel/aster-nix/src/net/socket/unix/stream/socket.rs:174]: Same issue

File: kernel/aster-nix/src/syscall/prctl.rs [hunk 4]
```diff
@@ -35,6 +35,14 @@ pub fn sys_prctl(option: i32, arg2: u64, arg3: u64, arg4: u64, arg5: u64) -> Res
 
             write_val_to_user(write_to_addr, &write_val)?;
         }
+        PrctlCmd::PR_GET_DUMPABLE => todo!(),
```
Comment (liqinggd) [inline kernel/aster-nix/src/syscall/prctl.rs:38]: `todo!()` will leads to a kernel panic here,  If we do not support it ,just leave it in  `_ => todo!(),`

File: kernel/aster-nix/src/net/socket/ip/datagram/mod.rs [hunk 5]
```diff
@@ -231,6 +237,32 @@ impl FileLike for DatagramSocket {
         Some(self)
     }
 
+    fn ioctl(&self, cmd: IoctlCmd, arg: usize) -> Result<i32> {
+        match cmd {
```
[the remark this answers is not in the document]
Comment (cqs21) [reply kernel/aster-nix/src/net/socket/ip/datagram/mod.rs:241]: It seems that those flags are only used by socket files, we'd better not provide default implementation for others.

File: kernel/aster-nix/src/net/socket/unix/stream/socket.rs (hunk 2 above)
Comment (cqs21) [reply kernel/aster-nix/src/net/socket/unix/stream/socket.rs:171]: To match the `FileLike` trait, it requires a `Uid`.

Comment (liqinggd) [reply kernel/aster-nix/src/net/socket/unix/stream/socket.rs:171]: The owner and group are the system's discretionary access control UID and GID.
Will using the PID as the UID in this context compromise the original access control mechanism? 
You need to conduct more detailed research to confirm whether returning the PID is appropriate.

**Comment by tatetian** [comment]
Please break this PR into atomic ones. This helps accelerate the review process.

**Pull request closed without merging**